codec-http/src/main/java/io/netty/handler/codec/http/HttpRequestEncoder.java

```diff
@@ -38,7 +39,7 @@ public boolean acceptOutboundMessage(Object msg) throws Exception {
     @Override
     protected void encodeInitialLine(ByteBuf buf, HttpRequest request) throws Exception {
         AsciiString method = request.method().name();
-        buf.writeBytes(method.array(), method.arrayOffset(), method.length());
```

Comment:
I left `arrayOffset()` just in case `AsciiString`/`ByteString` wraps a subregion of an existing byte array, which is not the case currently.  Non-zero `arrayOffset()` introduces more complexity in indexing and we are unsure about its performance impact.

Maybe we could remove `arrayOffset()` and let a user assume the array returned by `array()` will always have its first character at 0th index and have the exact same length with the `ByteString`, but I felt it might limit our chances to explore the potential room for improvements in the future.